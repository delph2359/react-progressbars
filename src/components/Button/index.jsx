import React, { Component } from "react";
import PropTypes from "prop-types";
import "./index.scss";

export default class Button extends Component {
  static propTypes = {
    text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    onClick: PropTypes.func.isRequired
  };

  render() {
    const { text, onClick } = this.props;
    return (
      <button className="button" onClick={onClick} data-testid="button">
        {text}
      </button>
    );
  }
}
