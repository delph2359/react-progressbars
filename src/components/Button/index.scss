.button {
  width: 100px;
  height: 40px;
  border-radius: 5px;
  border-color: #007aff;
  background: #fff;
  color: #007aff;
  font-size: 16px;
  margin: 5px;

  &:hover {
    background: #dfecfa;
  }
  &:focus {
    outline: none;
  }
}
