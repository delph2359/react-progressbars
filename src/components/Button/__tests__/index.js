import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import Button from "../";

describe("Button", () => {
  test("should display text correctly", () => {
    const { container, getByText } = render(
      <Button text={50} onClick={() => console.log("clicked")} />
    );

    expect(container).toMatchSnapshot();
    expect(getByText("50")).toBeInTheDocument();
  });
});
