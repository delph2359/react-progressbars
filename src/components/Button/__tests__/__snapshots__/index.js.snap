// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should display text correctly 1`] = `
<div>
  <button
    class="button"
    data-testid="button"
  >
    50
  </button>
</div>
`;
